gammapy/scripts/image_pipe.py

```diff
@@ -21,37 +28,152 @@ def image_pipe_main(args=None):
 
 class ImageAnalysis(object):
     """Gammapy 2D image based analysis.
+
+    center : `astropy.coordinates.SkyCoord`
+        Center of the image
+    energy_band : `~gammapy.utils.energy.Energy
+        Energy band for which we want to compute the image
+    offset_band :`astropy.coordinates.Angle`
+        Offset Band where you compute the image
+    data_store : `~gammapy.data.DataStore`
+        `DataStore` where are situated the events
     """
 
-    def __init__(self, config):
-        self.config = config
-
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def run(self):
-        """Run analysis chain."""
-        log.info('Running analysis ...')
-        print(self.config['general']['outdir'])
-        print(self.config)
-
-    def make_counts(self):
+    def __init__(self, center, energy_band, offset_band, data_store, counts_image=None, bkg_image=None):
+        self.data_store = data_store
```

Comment:
FYI: The ``SpectrumExtraction`` class takes a ``DataStore`` as input because when I wrote it there was no observation object. Probably passing around an observation object would be the better solution (at least short term)